content/blog/vscode-esp-idf-v1.8.0/index.md

```diff
@@ -0,0 +1,114 @@
+
+To use it in a ESP-IDF project `.vscode/launch.json`, you can add this configuration:
+
+```JSON
+{
+  "configurations": [
+    {
+      "type": "gdbtarget",
+      "request": "attach",
+      "name": "Eclipse CDT GDB Adapter"
+    }
+  ]
+}
+```
+
+After you build (**ESP-IDF: Build your Project**), flash (**ESP-IDF: Flash your Project**) you can go to menu `Run` and press `Start debugging`. You can check how to use the new debugger by following the [debug tutorial](https://github.com/espressif/vscode-esp-idf-extension/blob/master/docs/tutorial/debugging.md).
+
+{{< figure
+    default=true
+    src="img/debug-example.webp"
+    alt=""
+    caption=""
+    >}}
+
+## Advanced configuration of Eclipse CDT Debug Adapter
+
+There are many configurable options to customize the debug adapter if desired. You can view a deeper description in the [debugging documentation](https://github.com/espressif/vscode-esp-idf-extension/blob/master/docs/DEBUGGING.md).
+
+## ESP-IDF version switcher
+
+Whenever you use the **ESP-IDF: Configure ESP-IDF Extension** to install ESP-IDF in your system, these installation are saved in our extension as global state IDF setup. You can now use the **ESP-IDF: Select Current ESP-IDF Version** to set in your current project (workspace folder in Visual Studio Code language) the ESP-IDF version you desire to use. If you want to delete these IDF setups from the extension global state you can use the **ESP-IDF: Clear Saved ESP-IDF Setups**.
+
+{{< figure
+    default=true
+    src="img/idf-version-list.webp"
+    alt=""
+    caption=""
+    >}}
+
+The current ESP-IDF version for currently open ESP-IDF project is shown in the status bar items like this:
+
+{{< figure
+    default=true
+    src="img/idf-version-status-bar.webp"
+    alt=""
+    caption=""
+    >}}
+
+## Add ESP-IDF: idf.py reconfigure task command
+
+For code navigation the [Microsoft C/C++ Extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools) or [Clangd extension](https://marketplace.visualstudio.com/items?itemName=llvm-vs-code-extensions.vscode-clangd) can be used for C/C++ language support. These extensions provide feature such as syntax highlight, go to definition/declaration and other common language feature you would expect in an IDE. Many of these extensions rely on a `compile_commands.json` file, a JSON compilation database which consist of an array of “command objects”, where each command object specifies one way a translation unit is compiled in the project. More information about JSON compilation database can be found [here](https://clang.llvm.org/docs/JSONCompilationDatabase.html).
+
+The **ESP-IDF: idf.py reconfigure task** will execute ESP-IDF reconfigure task, which generate a compile_commands.json so the language extension can work properly. This is useful in case the user doesn't want to fully build the project but wants to see code navigation and other language features.
+
+## ESP-IDF Hints viewer support
```

Comment:
I am sure the users will have these questions in mind:

- What is `ESP-IDF Hints` and how does it work? Where do the hints come from?
- How is it different from code navigation? Does it also rely on `compile_commands.json`?